File: Projeto07_c++/Projeto07_c++/src/Monitor.cpp
#include "../include/Monitor.h"
#include <iostream>
using namespace std;

Monitor::Monitor(const string& nome, const string& email, const string& matricula, const string& curso,
                 const string& areaDeAtuacao, const vector<string>& disciplinasMinistradas, const vector<string>& disciplinasMonitoradas)
    : Aluno(nome, email, matricula, curso, ""), Professor(nome, email, areaDeAtuacao, ""), disciplinasMonitoradas(disciplinasMonitoradas) {
    for (const auto& disciplina : disciplinasMinistradas) {
        adicionarDisciplina(disciplina);
    }
}

void Monitor::gerarRelatorio() const {
    Aluno::gerarRelatorio();
    cout << "Disciplinas Ministradas: ";
    for (const auto& disciplina : getDisciplinasMinistradas()) {
        cout << disciplina << " ";
    }
    cout << endl;
    cout << "Disciplinas Monitoradas: ";
    for (const auto& disciplina : disciplinasMonitoradas) {
        cout << disciplina << " ";
    }
    cout << endl;
}

File: Projeto_8/src/models/Monitor.py
from src.models.Aluno import Aluno
from src.models.Professor import Professor
from src.models.TipoUsuario import TipoUsuario

class Monitor(Aluno, Professor):
    def __init__(self, 
                 nome='indefinido', 
                 email='indefinido', 
                 tipo=TipoUsuario.MONITOR, 
                 matricula='indefinida', 
                 curso='indefinido', 
                 disciplinasAluno=None, 
                 areaDeAtuacao='indefinida', 
                 disciplinasMinistradas=None, 
                 disciplinasMonitoradas=None):
        # Chamando o inicializador de Aluno e Professor usando super()
        super().__init__(nome, email, tipo, matricula, curso, disciplinasAluno or [])

        # Inicializando atributos exclusivos de Professor
        self._areaDeAtuacao = areaDeAtuacao
        self._disciplinasMinistradas = disciplinasMinistradas or []

        # Inicializando atributos exclusivos do Monitor
        self._disciplinasMonitoradas = disciplinasMonitoradas or []

    def listarDisciplinasMonitoradas(self):
        # Método para listar as disciplinas monitoradas pelo monitor
        if not self._disciplinasMonitoradas:
            return "Nenhuma disciplina está sendo monitorada no momento."
        return f"Disciplinas Monitoradas: {', '.join(self._disciplinasMonitoradas)}"
